src/CSET/operators/read.py

```diff
@@ -216,8 +217,81 @@ def _ensemble_callback(cube, field, filename: str):
 
 
-def _verify_paths(files: Union[Path, Iterable[Path]]):
-    """Verify file exists, warning otherwise."""
-    for file in iter_maybe(files):
-        if not file.is_file():
-            warnings.warn(f"File does not exist: {file}", RuntimeWarning, stacklevel=2)
+def _deterministic_callback(cube, field, filename):
+    """Give deterministic cubes a realization of 0.
+
+    This means they can be handled in the same way as ensembles through the rest
+    of the code.
+    """
+    # Only add if realization coordinate does not exist.
+    if not cube.coords("realization"):
+        cube.add_aux_coord(
+            iris.coords.AuxCoord(np.int32(0), standard_name="realization", units="1")
+        )
+
+
+def _lfric_normalise_callback(cube: iris.cube.Cube, field, filename):
+    """Normalise attributes that prevents LFRic cube from merging.
+
+    The uuid and timeStamp relate to the output file, as saved by XIOS, and has
+    no relation to the data contained. These attributes are removed.
+
+    The um_stash_source is a list of STASH codes for when an LFRic field maps to
+    multiple UM fields, however it can be encoded in any order. This attribute
+    is sorted to prevent this. This attribute is only present in LFRic data that
+    has been converted to look like UM data.
+    """
+    # Remove unwanted attributes.
+    cube.attributes.pop("timeStamp", None)
+    cube.attributes.pop("uuid", None)
+
+    # Sort STASH code list.
+    stash_list = cube.attributes.get("um_stash_source")
+    if stash_list:
+        # Parse the string as a list, sort, then re-encode as a string.
+        cube.attributes["um_stash_source"] = str(sorted(ast.literal_eval(stash_list)))
+
+
+def _check_input_files(input_path: Path | str, filename_pattern: str) -> Iterable[Path]:
+    """Get an iterable of files to load, and check that they all exist.
+
+    Arguments
+    ---------
+    input_path: Path | str
+        Path to an input file or directory. The path may itself contain glob
+        patterns, but unlike in shells it will match directly first.
+
+    filename_pattern: str
+        Shell-style glob pattern to match inside the input directory.
+
+    Returns
+    -------
+    Iterable[Path]
+        A list of files to load.
+
+    Raises
+    ------
+    FileNotFoundError:
+        If the provided arguments don't resolve to at least one existing file.
+    """
+    # Convert string paths into Path objects.
+    if isinstance(input_path, str):
+        input_path = Path(input_path)
+
+    # Get the list of files in the directory, or use it directly. Error if no
+    # files found.
+    if input_path.is_dir():
+        files = tuple(sorted(input_path.glob(filename_pattern)))
+        if len(files) == 0:
+            raise FileNotFoundError(
+                f"No files found matching glob {filename_pattern} within {input_path}"
+            )
+    elif input_path.is_file():
+        files = (input_path,)
```

Comment:
will this raise error if file doesn't exist and `is_file` is invoked? Just checking I follow the logic

Reply:
Only if the input_path doesn't contain a globbing pattern that finds some files. The logic should follow this flowchart.

```mermaid
flowchart TD
    A[input_path] --> C{exists and directory?}
    C --> |Yes| D{contains files?}
    C --> |No| F{exists and file?}
    D --> |Yes| B[return list of files]
    D --> |No| E[raise FileNotFoundError]
    F --> |No| H{glob matches files?}
    F --> |Yes| G[return list of one file]
    H --> |Yes| B
    H --> |No| E
```

Reply:
Thanks for diagram, that helps. I guess my point if user points at one file only (left hand side of it), and it doesn't exist, shouldn't that also raise `FileNotFoundError`? I guess iris itself would return that in any case.